cumulusci/utils/salesforce/count_sobjects.py

```diff
@@ -0,0 +1,46 @@
+import typing as T
+
+from simple_salesforce import Salesforce
+
+from cumulusci.utils.http.multi_request import CompositeParallelSalesforce
+from cumulusci.utils.iterators import partition
+
+
+class ObjectCount(T.NamedTuple):
+    counts: T.Dict[str, int]
+    transport_errors: T.Sequence[dict]  # e.g. couldn't reach server
+    salesforce_errors: T.Sequence[dict]  # e.g. 404, 401
+
+
+def count_sobjects(sf: Salesforce, objs: T.Sequence[str]) -> ObjectCount:
+    """Quickly count SObjects using SOQL and Parallelization"""
+    with CompositeParallelSalesforce(sf, max_workers=8, chunk_size=5) as cpsf:
+        responses, transport_errors = cpsf.do_composite_requests(
+            (
+                {
+                    "method": "GET",
+                    "url": f"/services/data/v{sf.sf_version}/query/?q=select count() from {obj}",
+                    "referenceId": f"ref{obj}",
+                }
+                for obj in objs
+            )
+        )
+        salesforce_errors, successes = partition(
+            lambda response: response["httpStatusCode"] == 200, responses
+        )
+
+        transport_errors = tuple(error._asdict() for error in transport_errors)
```

Comment:
`_asdict()` isn't public API. Acknowledging that it's less elegant, how about
```suggestion
        transport_errors = tuple({"exception": error.exception, "request": error.request} for error in transport_errors)
```

Reply:
I promise I'm not still reading CumulusCI pull requests closely, but I still get notifications and saw this go by. `_asdict` _is_ documented public API, despite the underscore: https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict -- namedtuple uses the underscore for its public methods to avoid the potential for collision with namedtuple field names

Reply:
Oh good grief. Thanks for the heads-up, @davisagli.